### PR TITLE
Add more statuses to browse list

### PR DESCRIPTION
**Pull request by quis**
img_2790_comp

--

This commit extends/rewrites the work done in #178. Before, statuses were just lines of text, which could be bold or not bold.

--

Now, there are four types of status message:
- quiet
- default
- angry
- happy

Each of these types has one mandatory part:
- title

…and two optional parts:
- hint
- list (eg to show the user which tasks they have completed so far)

Furthermore, it is now possible to have multiple status blocks per item, eg to show tasks completed and tasks not completed.

--

**[Preview the documentation](http://quis.github.io/digitalmarketplace-frontend-toolkit/browse-list.html)**

image

``` Jinja
{%
  with
  items = [
    {
        "link": "#nope", 
        "statuses": [
            {
                "type": "quiet", 
                "title": "You haven\u2019t started this yet"
            }
        ], 
        "title": "Apply to G-Cloud 11"
    }, 
    {
        "body": "Provide teams to build and support digital outcomes.", 
        "link": "#nope", 
        "statuses": [
            {
                "type": "happy", 
                "list": [
                    "digital outcomes", 
                    "digital specialists"
                ], 
                "title": "You are submitting"
            }, 
            {
                "type": "quiet", 
                "list": [
                    "user research studios", 
                    "participant recruitment"
                ], 
                "title": "You are not submitting"
            }
        ], 
        "title": "Apply to provide teams for digital outcomes"
    }, 
    {
        "body": "Provide individual specialists to work on digital services, programmes or projects.", 
        "link": "#nope", 
        "statuses": [
            {
                "title": "You have have completed this service.", 
                "hint": "You can edit it until the deadline"
            }
        ], 
        "title": "Apply to provide digital specialists"
    }, 
    {
        "body": "Put yourself through the tender window", 
        "link": "#nope", 
        "statuses": [
            {
                "type": "angry", 
                "title": "You must do this before any services can be submitted"
            }
        ], 
        "title": "Make supplier declaration"
    }
]
%}
  {% include "toolkit/browse-list.html" %}
{% endwith %}
```

--

**Bump version to 12.0.0**

This is a breaking change because it changes the expected parameters of browse list’s status.

### CODE REVIEW

File: pages_builder/pages/browse-list.yml [hunk 1]
```diff
@@ -26,31 +26,42 @@ examples:
...
+            title: You haven’t started this yet
+      -
         title: Apply to provide teams for digital outcomes
         link: '#nope'
         body: Provide teams to build and support digital outcomes.
         statuses:
           -
-            message: You have completed this service.
-            important: True
+            type: happy
```
Comment (TheDoubleK) [inline pages_builder/pages/browse-list.yml]: In the generated patterns the `title` ends up at the bottom of the lists for some reason, which isn't as nice as in the YAML here where the title is at the top, e.g.: 
screen shot 2015-11-16 at 08 49 42

It's not the end of the world, but not quite as clear what's going on.

File: pages_builder/pages/browse-list.yml [hunk 2]
```diff
@@ -26,31 +26,42 @@ examples:
...
           -
-            message: You have not completed this service.
+            title: You have have completed this service.
+            hint: You can edit it until the deadline
       -
-        title: Apply to provide software as a service
+        title: Make supplier declaration
         link: '#nope'
-        body: G-Cloud 10 is open for applications until 1 October 2019
+        body: Put yourself through the tender window
```
Comment (TheDoubleK) [inline pages_builder/pages/browse-list.yml]: :)

**Comment by TheDoubleK** [comment]
Looks nice.

Would it be less breaking to stick with `message` instead of `title` as the key for statuses?

File: pages_builder/pages/browse-list.yml (hunk 1 above)
Comment (quis) [inline pages_builder/pages/browse-list.yml]: Agree that this would be nice to have at some point.

**Pull request merged**